sympy/calculus/tests/test_euler.py

```diff
@@ -62,3 +62,13 @@ def test_euler_high_order():
     w = Symbol('w')
     L = D(x(t, w), t, w)**2/2
     assert euler(L) == [Eq(D(x(t, w), t, t, w, w), 0)]
+
+
+def test_euler_when_L_out_turn_to_zero():
+    x = Symbol('x')
+    y = Symbol('y')
+    z = Symbol('z')
+    f = Function('f')
+    h = Function('h')
+    L = f.diff(x)*h.diff(z)
+    assert euler(L, f(x,y), [x,y]) == 0
```

Comment:
This function normally returns an `Equality` rather than an `Expr`.

Reply:
I think we can't make it to Expr since 
`euler(f.diff(x)*h.diff(z), f(x,y), [x,y]) `
is equivalent to 
`euler(1, f(x,y), [x,y]) `
when we evaluate L inside that it returns number so I think it's not possible to retain L in `Expr`

Reply:
```
>>> L = f.diff(x)*h.diff(z)
>>> print(L)
1
```

Reply:
@oscarbenjamin Any views on the above comments from @MaqOwais

Reply:
I don't understand the comments above but return types should be kept consistent.